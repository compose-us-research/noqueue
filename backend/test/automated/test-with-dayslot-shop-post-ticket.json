{
  "start": "2020-11-20",
  "end": "2020-11-22",
  "contact": {
    "name": "Max Mustermann",
    "email": "max@mustermann.example.org"
  }
}
