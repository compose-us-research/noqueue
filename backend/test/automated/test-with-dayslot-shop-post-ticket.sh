#!/bin/bash

source setup-dayslot.sh

DATA=$(cat test-with-dayslot-shop-post-ticket.json)
RESULT=<(curl -s -w "%{http_code}" $BASE_URL/shop/$SHOP_NAME/ticket/ -X POST -H "content-type: application/json" -H "accept: application/json" --data "$DATA")
EXPECTED=<(cat test-with-dayslot-shop-post-ticket.expected)

testIt "Posting ticket" $RESULT $EXPECTED
