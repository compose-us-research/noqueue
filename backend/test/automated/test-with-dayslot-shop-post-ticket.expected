{"code":20,"message":"InvalidRequest - no availability for this ticket"}
___HEADER___
400
